### PR TITLE
NPE fixes and v1.6 release

### PR DESCRIPTION
**Pull request by suomiy**
This change is [Reviewable](https://reviewable.io/reviews/matobet/movirt/194)

### CODE REVIEW

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java [hunk 1]
```diff
@@ -225,7 +225,6 @@ public void onLoaderReset(Loader<Cursor> loader) {
         // do nothing
     }
 
-    @UiThread
     public void renderVm(Vm vm) {
         if (!vm.isSnapshotEmbedded()) {
             getActivity().setTitle(String.format(VM_DETAILS, vm.getName()));
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java]: is this safe outside of ui thread?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java [hunk 2]
```diff
@@ -250,7 +249,6 @@ public void renderVm(Vm vm) {
         }
     }
 
-    @UiThread
     public void renderHost(Host host) {
         if (host != null) {
             hostButton.setText(host.getName());
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java]: is this safe outside of ui thread?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java [hunk 3]
```diff
@@ -261,12 +259,10 @@ public void renderHost(Host host) {
         }
     }
 
-    @UiThread
     public void renderCluster(Cluster cluster) {
         clusterView.setText(getString(R.string.two_separated_strings, cluster.getName(), cluster.getVersion()));
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java]: is this safe outside of ui thread?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java [hunk 4]
```diff
@@ -261,12 +259,10 @@ public void renderHost(Host host) {
...
     }
 
-    @UiThread
     public void renderDataCenter(DataCenter dataCenter) {
         dataCenterView.setText(getString(R.string.two_separated_strings, dataCenter.getName(), dataCenter.getVersion()));
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java]: is this safe outside of ui thread?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java [hunk 5]
```diff
@@ -226,23 +226,20 @@ void start() {
     }
 
     @OptionsItem(R.id.action_stop)
-    @UiThread
     void stop() {
         DialogFragment confirmDialog = ConfirmDialogFragment
                 .newInstance(ACTION_STOP_VM, getString(R.string.dialog_action_stop_vm));
         confirmDialog.show(getFragmentManager(), "confirmStopVM");
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java]: is this safe outside of ui thread?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java [hunk 6]
```diff
@@ -226,23 +226,20 @@ void start() {
...
     }
 
     @OptionsItem(R.id.action_reboot)
-    @UiThread
     void reboot() {
         DialogFragment confirmDialog = ConfirmDialogFragment
                 .newInstance(ACTION_REBOOT_VM, getString(R.string.dialog_action_reboot_vm));
         confirmDialog.show(getFragmentManager(), "confirmRebootVM");
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java]: is this safe outside of ui thread?

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java [hunk 7]
```diff
@@ -226,23 +226,20 @@ void start() {
...
     }
 
     @OptionsItem(R.id.action_cancel_migration)
-    @UiThread
     void cancelMigration() {
         DialogFragment confirmDialog = ConfirmDialogFragment
                 .newInstance(ACTION_STOP_MIGRATE_VM,
```
Comment (jelkosz) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java]: is this safe outside of ui thread?

**Comment by jelkosz** [comment]
@suomiy just questions about safeness - what guarantees that that code is being run on ui thread? If you have a good reason to believe than Im ok with the change

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java (hunk 1 above)
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailGeneralFragment.java]: This is safe if we call these functions from onLoadFinished which was [initialized](http://stackoverflow.com/questions/30825534/which-threads-call-oncreateloader-and-onloadfinished) in UiThread. This is exactly what we do (We call init from @AfterViews), which is called from UiThread.

File: moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java (hunk 5 above)
Comment (suomiy) [inline moVirt/src/main/java/org/ovirt/mobile/movirt/ui/vms/VmDetailActivity.java]: I couldn't find documentation for this (same as the 2nd comment), but i found many examples on the Internet which use @OptionsItem (ie onOptionsItemSelected ) as UiThread (We also do). I also debugged it and concluded it runs on the main thread and we do not need @UiThread annotation here.

**Comment by suomiy** [comment]
@jelkosz I guess there is no real gurantee (viz. comments). I tested it and it ran correctly. 

Other possibility is to add `(propagation = UiThread.Propagation.REUSE) `to those deleted `@UiThread` annotations. We also have old code which doesn't use these annotations and I would prefer not to put superfluous code there if we do not need it. 
It behaves correctly without them and forgetting propagation option can lead to bugs.

**Comment by suomiy** [comment]
@jelkosz I believe it is safe :), and also if we call accidentally something from background thread in the future, we will get exceptions very soon.

**Comment by jelkosz** [comment]
@suomiy ok, seems legit :) merging

**Pull request merged**